### ISSUE TITLE
Deprecate the `headers` and `received_headers` attributes of `IRequest` and implementations of that interface

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6704]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6704|
|Type|enhancement|
|Created|2013-08-26 16:48:05Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-old-headers-6704|

These attributes were made into cumbersome aliases (for `responseHeaders` and `requestHeaders` respectively) some time ago.  We should nudge everyone towards the new interface so we can actually get rid of the old one and its support code some day.

<details><summary>Searchable metadata</summary>

```
trac-id__6704 6704
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_deprecate_old_headers_6704 branches/deprecate-old-headers-6704
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1377535685000000 1377535685000000
changetime__1378919192000000 1378919192000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [39752]) Branching to 'deprecate-old-headers-6704'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set owner to @exarkun<br>@wallrj set status to `new`|
|-|-|

Thanks exarkun,

This all looks fine.

Notes:
 * Merges cleanly
 * t.web tests pass
 * Checked that there aren't any remaining uses of headers or
   request_headers outside of the tests.
 * Checked that the warnings are shown with PYTHONWARNINGS=default python
 * I wondered whether it might be better to use addRawHeader instead
   of setRawHeaders so as not to replace headers that might already
   have been present. But I see that the _DictHeaders thing works that
   way, so ok.
 * Re-ran the pyflakes builder and it went green
 * Re-ran the twistedchecker builder and it had fewer warnings.

Points:
 1. [branches/deprecate-old-headers-6704/twisted/web/test/test_http.py](branches/deprecate-old-headers-6704/twisted/web/test/test_http.py)
    1. `sub` needs documentation (https://buildbot.twistedmatrix.com/builders/twistedchecker/builds/1275/steps/run-twistedchecker/logs/new%20twistedchecker%20errors)


Please merge after adding that missing docstring.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [39951]) Merge deprecate-old-headers-6704

Author: exarkun
Reviewer: rwall
Fixes: [#6704](https://github.com/twisted/twisted/issues/6704)

The old-style header attributes of `twisted.web.iweb.IRequest`, `received_headers` and
`headers` are now deprecated in favor of the more featureful `requestHeaders` and
`responseHeaders` attributes.
